### ISSUE TITLE
confirmation toasts: focus or alert?

### ISSUE THREAD

**Issue by plweil** [dequelabs/pattern-library#60]
I see that confirmation toasts are using focus instead of an aria alert role. I'm hearing from other developers (e.g., on the A11Y Slack channel) that shifting focus is "bad UX", and they advocate using aria live regions (e.g., alerts) that disappear after a certain time instead. I honestly don't know which way is better. What is the rationale for relying on focus? Is shifting focus really ok or better for the user than a live region?

**Comment by schne324** [comment]
I think its a matter of taste / context. I'm not on that slack channel so I didn't get to read the full statement about shifting focus being "bad UX" but I will say in some cases the toast message could contain interactive elements link links, in which case I'd say focusing it is the way to go.  On the other hand, if it is a simple confirmation message, `role="alert"` is the way to go.  Either way, this isn't an issue with the pattern library itself (the [toast component](https://github.com/dequelabs/pattern-library/tree/master/lib/components/toasts) is only styles and it is up to the implementer to make this call).  Nothing about using the deque-pattern-library toast component forces you focus the toast. Nothing about using the deque-pattern-library toast component forces you to add `role="alert"`.

On the demo site, we made the decision to not auto-hide the toasts and focus them which makes the user deliberately close them.  This is by no means the only way to use the toasts.  Perhaps we could update the demos to show an example using `role="alert"`.

What I'm going to do is update the wiki directing implementers of the toast component to make a judgement call and present them with both options.

Thanks for the feedback!

**Comment by schne324** [comment]
I've also created an internal ticket for our demo site to consider updating the demos with a `role="alert"` example.

**Comment by plweil** [comment]
Thanks a lot for the quick response. I’ve Been going back and forth on this, trying to stay within the specs and doing the best thing for the user. I agree  that this  calls for some common sense and depends a lot on the specific context. Thanks again.
